src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -367,114 +377,45 @@ public static boolean takeStackInSlot(final IInventory sendingInv, final IInvent
      * @return True if item is swapped, otherwise false
      */
     public static boolean takeStackInSlot(
-                                           @Nullable final IInventory sendingInv, @Nullable final IInventory receivingInv,
+                                           @Nullable final IItemHandler sendingInv, @Nullable final IItemHandler receivingInv,
                                            final int slotID, final int amount, final boolean takeAll)
     {
         if (receivingInv != null && sendingInv != null && slotID >= 0 && amount >= 0)
         {
             // gets itemstack in slot, and decreases stacksize
-            @Nullable ItemStack stack = sendingInv.decrStackSize(slotID, amount);
-            // stack is null if no itemstack was in slot
-            if (stack != null)
+            @Nullable ItemStack stack = sendingInv.extractItem(slotID, takeAll ? 64 : amount, true);
+            @Nullable ItemStack leftover = ItemHandlerHelper.insertItemStacked(receivingInv, stack, true);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'leftover' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)